File: dotnet/Vaas/src/Vaas/Options/ForFileOptions.cs
namespace Vaas.Options;

public class ForFileOptions
{
    public bool UseCache { get; init; } = true;
    public bool UseHashLookup { get; init; } = true;
    public string? VaasRequestId { get; init; }

    public static ForFileOptions From(VaasOptions options)
    {
        return new ForFileOptions
        {
            UseCache = options.UseCache,
            UseHashLookup = options.UseHashLookup,
            VaasRequestId = null,
        };
    }
}

File: typescript/src/options/ForFileOptions.ts
import { VaasOptions } from "./VaasOptions";

export class ForFileOptions {
  public useCache: boolean;
  public useHashLookup: boolean;
  public vaasRequestId?: string;

  constructor(options?: {
    useCache?: boolean;
    useHashLookup?: boolean;
    vaasRequestId?: string;
  }) {
    this.useCache = options?.useCache ?? true;
    this.useHashLookup = options?.useHashLookup ?? true;
    this.vaasRequestId = options?.vaasRequestId;
  }

  public static from(options: VaasOptions): ForFileOptions {
    return new ForFileOptions({
      useCache: options.useCache,
      useHashLookup: options.useHashLookup,
    });
  }
}
